paddle/fluid/imperative/layer.h

```diff
@@ -115,12 +123,16 @@ class VarBase {
   int pre_op_out_idx_;
 
   framework::VarDesc* var_desc_;
-  framework::Variable* var_;
-  framework::Variable* grads_;
+
+  std::shared_ptr<framework::Variable> var_;
```

Comment:
delete in destructor and don't use share_ptr now?

Reply:
Done